### PR TITLE
Send "admin" and "add_groups" properties

### PR DESCRIPTION
**Pull request by brianhelba**
* Send "admin" and "add_groups" properties
* Dynamically set "require_activation"
* Add new tests from upstream
* Don't render the secret string as a password
  * This may need to be copied to Discourse in some cases

### CODE REVIEW

**Comment by brianhelba** [comment]
@msmolens A code review would be helpful, but I already tested this on our instance.

**Comment by brianhelba** [comment]
I just realize this isn't tested, and may actually break the existing tests. I'll remedy.

File: server/api.py [hunk 1]
```diff
@@ -86,10 +87,17 @@ def login(self, params):
...
             'email': user['email'],
-            'external_id': user['_id'],
+            'external_id': str(user['_id']),
             'username': user['login'],
             'name': '%s %s' % (user['firstName'], user['lastName']),
-            'require_activation': 'true' if requireActivation else 'false'
+            'require_activation': 'true' if requireActivation else 'false',
+            'admin': 'true' if user['admin'] else 'false',
+            'add_groups': ','.join(
+                group['name']
```
Comment (msmolens) [inline server/api.py]: Does the group name have to match the "Name" group field on Discourse as opposed to the "Full Name" field? If so, it looks like there will be a mismatch with the current configuration.

See https://github.com/discourse/discourse/blob/dd101e4f1dc556ba1fcfde5e95c39665bff9792d/app/models/discourse_single_sign_on.rb#L113.

**Comment by brianhelba** [comment]
@msmolens You're right about groups. When I tested this, I think I was actually seeing the result of temporarily UI inconsistency in Discourse when I both "member" and an "owner" of the group. I added a comment to clarify the situation. Group mapping is still useful as a feature, so I'm leaving it intact, even if it doesn't help our own usage.

Otherwise, PTAL.

File: plugin_tests/discourse_sso_test.py [hunk 2]
```diff
@@ -158,18 +171,17 @@ def testDiscourseLogin(self):
              'require_activation'))
```
Comment (msmolens) [inline plugin_tests/discourse_sso_test.py]: This list should be updated to include `admin` and `add_groups`.

File: plugin_tests/discourse_sso_test.py [hunk 3]
```diff
@@ -199,11 +211,13 @@ def testDiscourseLogin(self):
             ('nonce', 'email', 'external_id', 'username', 'name',
              'require_activation'))
```
Comment (msmolens) [inline plugin_tests/discourse_sso_test.py]: This list should be updated as well.

File: plugin_tests/discourse_sso_test.py [hunk 4]
```diff
@@ -199,11 +211,13 @@ def testDiscourseLogin(self):
...
-        self.assertEqual(parsed['external_id'][0], str(self.user['_id']))
-        self.assertEqual(parsed['username'][0], 'user1login')
-        self.assertEqual(parsed['name'][0], 'First Last')
+        self.assertEqual(parsed['email'][0], 'user2@email.com')
+        self.assertEqual(parsed['external_id'][0], str(adminUser['_id']))
+        self.assertEqual(parsed['username'][0], 'user2login')
+        self.assertEqual(parsed['name'][0], 'Admin Last')
         self.assertEqual(parsed['require_activation'][0], 'false')
+        self.assertEqual(parsed['admin'][0], 'true')
+        self.assertNotIn('add_groups', parsed)
```
Comment (msmolens) [inline plugin_tests/discourse_sso_test.py]: Why is `add_group` not present for the admin user?

Comment (brianhelba) [reply plugin_tests/discourse_sso_test.py]: In this test case, the admin user is a member of no groups. When that gets URL-encoded, it is logically `add_groups=`, which actually gets emitted as an empty string, due to the internal rules of query strings.

**Comment by brianhelba** [comment]
@msmolens Fixed.

File: plugin_tests/discourse_sso_test.py (hunk 4 above)
Comment (msmolens) [reply plugin_tests/discourse_sso_test.py]: I see, thanks!

**Pull request merged**